### ISSUE TITLE
pandoc error: "--chapters has been removed."

### ISSUE THREAD

**Issue by mwyczalkowski** [hadley/ggplot2-book#134]
Trying to make book on OS X 10.12.6, get failure while doing `make`:

```
Rscript book/render-tex.R data-manip.rmd
--chapters has been removed. Use --top-level-division=chapter instead.
Try pandoc --help for more information.
Error: pandoc document conversion failed with error 2
Execution halted
make: *** [book/tex/data-manip.tex] Error 1
```

Using `pandoc` v. 2.0.5

**Comment by edwardslee** [comment]
I got the same error just now and got past it by replacing the code in the `render-tex.R` file in `ggplot2-book/book/` folder with this:

```library("methods") # avoids weird broom error
library("rmarkdown")

tex_chapter <- function (chapter = NULL, latex_engine = c("xelatex", "pdflatex",
                                                          "lualatex"), code_width = 65) {
  options(digits = 3)
  set.seed(1014)
  latex_engine <- match.arg(latex_engine)
  rmarkdown::output_format(rmarkdown::knitr_options("html", chapter), rmarkdown::pandoc_options(to = "latex",
                                                                                  from = "markdown_style", ext = ".tex", args = c("--top-level-division=chapter",                                                                                                                            rmarkdown::pandoc_latex_engine_args(latex_engine))), clean_supporting = FALSE)
}

path <- commandArgs(trailingOnly = TRUE)
# command line args should contain just one chapter name
if (length(path) == 0) {
  message("No input supplied")
} else {
  base <- tex_chapter()
  base$knitr$opts_knit$width <- 67
  base$pandoc$from <- "markdown"

  rmarkdown::render(path, base, output_dir = "book/tex", envir = globalenv(), quiet = TRUE)
}
```

The `oldbookdown::tex_chapter()` call seems to be using a deprecated pandoc option.

**Comment by mwyczalkowski** [comment]
Thank you! That helped.

Running into additional compilation issues, will post on new thread.

**Comment by edwardslee** [comment]
I'm not quite sure what's going on after a quick look. The issue is in lines 684-692 in `toolbox.rmd`:

```
library(USAboundaries)
c18 <- us_boundaries(as.Date("1820-01-01"))
c18df <- fortify(c18)
head(c18df)
 
ggplot(c18df, aes(long, lat)) + 
   geom_polygon(aes(group = group), colour = "grey50", fill = NA) +
   coord_quickmap()
```

If you look at the columns of `c18df`, there's no column named `group`.
```
#> names(c18df)
 [1] "id_num"     "name"       "id"         "version"    "start_date" "end_date"   "change"     "citation"   "start_n"   
[10] "end_n"      "area_sqmi"  "terr_type"  "full_name"  "abbr_name"  "name_start" "state_abbr" "state_name" "state_code"
[19] "geometry" 
```
So the call in `geom_polygon()`, `aes(group = group)` would throw an error. 

One way to get around this would be to just comment out this block and then try to compile the book without it.

**Comment by hlynurhallgrims** [comment]
It seems to me that this has to do with changes in the USAboundaries package between version 0.2.0 and 0.3.0. I tried downgrading to 0.2.0 and it worked.

`devtools::install_version(package = "USAboundaries", version = "0.2.0")`

**Comment by jgendrinal** [comment]
did someone already put @edwardslee's solution into a pull request?

Still had to change my render-tex.R file for it to work.

**Comment by rfaelens** [comment]
In case anyone stumbles upon this when building a `tuftebook` rmarkdown: this is solved in the latest version of tufte on github.

**Comment by hadley** [comment]
Fixed by #142
